### PR TITLE
Set up bracketing for spatial solvers.

### PR DESCRIPTION
**Pull request by Vilin97**
This is in preparation for the spatial SSA RSSACRDirect.

### CODE REVIEW

File: src/spatial/bracketing.jl [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    end
+    nothing
+end
+
+### convenience functions for LowHigh ###
+function setindex!(low_high::LowHigh, val::LowHigh, i)
+    low_high.low[i] = val.low
+    low_high.high[i] = val.high
+    val
+end
```
[the remark this answers is not in the document]
Comment (isaacsas) [reply src/spatial/bracketing.jl:30]: Also, do you need a corresponding `getindex`?

File: src/spatial/reaction_rates.jl [hunk 2]
```diff
@@ -52,16 +52,21 @@ end
 
 update rates of all reactions in rxs at site
 """
-function update_rx_rates!(rx_rates::RxRates{F, M}, rxs, integrator,
-                          site) where {F, M <: AbstractMassActionJump}
-    u = integrator.u
+function update_rx_rates!(rx_rates::RxRates{F, M}, rxs, u::AbstractMatrix, integrator,
```
Comment (isaacsas) [inline src/spatial/reaction_rates.jl:55]: If `F` and `M` aren't constrained why make it parametric like this? i.e. why not just take `rx_rates::RxRates`?

**Review by isaacsas — COMMENTED** [review]
Would it make sense to have RSSA/RSSACR store the high/low values together in general? i.e. is it likely that whenever we need high we may shortly thereafter need low, and so should just return both? I wonder if that might give a perf bump by storing them together in memory.

File: src/spatial/bracketing.jl (hunk 1 above)
[the remark this answers is not in the document]
Comment (Vilin97) [reply src/spatial/bracketing.jl:30]: `LowHigh` can store a number as well, this function just won't be called in that case. `setindex!` is called whenever an assignment like `low_high[i] = val` is done. 

I added a `getindex`, good suggestion.

File: src/spatial/reaction_rates.jl (hunk 2 above)
Comment (Vilin97) [reply src/spatial/reaction_rates.jl:55]: There was no reason but oversight. Good catch.

**Comment by Vilin97** [comment]
I do like the little handy `LowHigh` struct. It makes RSSA code look similar to normal code, which is good for maintainability. I'm not sure about performance, that would need to be measured. My instinct is that we would not win much performance by switching non-spatial RSSA-based SSAs to `LowHigh`.

**Review by isaacsas — APPROVED** [review]

**Pull request merged**